bsp/src/main.rs

```diff
@@ -1,3 +1,21 @@
-fn main() {
-    println!("Hello, world!");
+use storage_kit::manager::StorageKitBuilder;
+use tracing_subscriber::util::SubscriberInitExt;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let filter = tracing_subscriber::EnvFilter::try_from_default_env()?;
+    tracing_subscriber::FmtSubscriber::builder()
+        .with_env_filter(filter)
+        .finish()
+        .try_init()?;
+
+    let mut storage_kit_manager = StorageKitBuilder::new()
+        .start()
```

Comment:
> I would rename this to build since below is when we actually start the process with start_as_bsp().

My concern is that build it is a bit misleading since it actually starts 2 threads (for now) in the background.

> I would have the builder be able to specify run_as(Roles::BSP), and then when you start below, it will execute all processes related to the roles you chose in the builder.

Is there a reason why would someone want to start as multiple roles at once? :D

I agree it is cleaner to make the builder just build. Will refactor this.

Reply:
I guess it’s either they run the executable twice, once as a BSP and the other as an MSP. Or they run as both for the same executable.